javacutil/src/org/checkerframework/javacutil/AnnotationBuilder.java

```diff
@@ -153,6 +154,39 @@ public AnnotationMirror build() {
         return new CheckerFrameworkAnnotationMirror(annotationType, elementValues);
     }
 
+    /**
+     * Copies every element value from the given annotation. If an element in the given annotation
+     * doesn't exist in the annotation to be built, an error would be raised unless the element is
+     * specified in {@code ignorableElements}.
+     *
+     * @param valueHolder the annotation that holds the values to be copied
+     * @param ignorableElements the elements that can be safely dropped
+     */
+    public AnnotationBuilder copyElementValuesFromAnnotation(
+            AnnotationMirror valueHolder, String... ignorableElements) {
+        for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> eltValToCopy :
+                valueHolder.getElementValues().entrySet()) {
+            Name eltNameToCopy = eltValToCopy.getKey().getSimpleName();
+            if (Arrays.asList(ignorableElements).contains(eltNameToCopy.toString())) {
+                continue;
+            }
+            boolean hasElement = false;
+            for (ExecutableElement elt :
```

Comment:
Can't you simplify the rest of this by calling `AnnotationBuilder setValue(CharSequence key, Object value)`? That method finds the right method and adds the value, additionally ensures that it's of the right type, and outputs the same error message.

Reply:
It's been too long that I forgot why I did this. I tried to use `setValue` but the `checkSubtype` method can't be passed.  I was trying to replace this chunk of code with the following:
```java
setValue(eltNameToCopy, (Object) eltValToCopy.getValue());
```
It keeps saying that `given value differs from expected; found: com.sun.tools.javac.code.Attribute.Array; expected: java.lang.String[]`

Do you have any insight on what happens here?